modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/KeyValueModifyPlan.java

```diff
@@ -130,8 +136,22 @@ public <RowT> AsyncCursor<InternalSqlRow> execute(
 
                     UpdatableTable updatableTable = execTable.updatableTable();
 
+                    RowT row = rowSupplier.get();
+
+                    RelDataType rowType = table().getRowType(ctx.getTypeFactory());
+
+                    Supplier<RowSchema> schemaSupplier = () -> {
+                        if (rowSchema != null) {
+                            return rowSchema;
+                        }
+                        rowSchema = rowSchemaFromRelTypes(RelOptUtil.getFieldTypeList(rowType));
+                        return rowSchema;
+                    };
+
+                    row = validateCharactersOverflowAndTrimIfPossible(rowType, ctx.rowHandler(), row, schemaSupplier);
```

Comment:
not sure why do you do validation outside of the UpdateableTable

Reply:
got it, done